### PR TITLE
Implement survey take view

### PR DESCRIPTION
**Pull request by jonasreholt**
Issue: #4 

This PR creates a survey taking screen.

# Testing
1. A small amount of automated tests has been added and does pass
2. Manually tested the visuals and navigation

# UI
## Opening a survey
Screenshot from 2024-05-25 00-27-50

## Going to next page
Screenshot from 2024-05-25 00-28-05

## Going back again
(Note before I went forward I chose some answers)
Screenshot from 2024-05-25 00-28-13

## Clicking quick or finish
Screenshot from 2024-05-25 00-28-22

### CODE REVIEW

File: src/scivu/scivu/ViewModels/ExitSurveyViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public ExitSurveyViewModel()
+    {
+        YesCommand = ReactiveCommand.Create(() =>
+        {
+            Console.WriteLine("Hej from yesh");
+            return (object?)true;
+        });
+        NoCommand = ReactiveCommand.Create(() =>
+        {
+            Console.WriteLine("Hej from no");
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/ExitSurveyViewModel.cs:21]:
```suggestion
```

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs [hunk 2]
```diff
@@ -1,21 +1,92 @@
...
+        QuitCommand = ReactiveCommand.CreateFromTask(async () =>
+        {
+            var dialog = new ExitSurveyViewModel();
+
+            var result = await ShowDialog.Handle(dialog);
+
+            if (result)
+            {
+                // Do the quit
+                Console.WriteLine("Quitting survey");
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:54]:
```suggestion
```

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs [hunk 3]
```diff
@@ -27,4 +98,125 @@ public bool IsLastPage
...
+                {
+                    AnswerType.Text => throw new NotImplementedException(t.ToString()),
+                    AnswerType.Scale => new ScaleQuestionViewModel(question),
+                    AnswerType.MultipleChoice => throw new NotImplementedException(t.ToString()),
+                    _ => throw new UnreachableException($"Setting questions with `{t}`")
+                });
+        }
+    }
+
+    private void SaveQuestionAnswers()
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:159]:
```suggestion
    private void SaveQuestionResults()
```

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs [hunk 4]
```diff
@@ -27,4 +98,125 @@ public bool IsLastPage
...
+    {
+        var currentAnswerList = _results[_resultIdx];
+        // Clear any possible old answers as we are about to overwrite anyway
+        currentAnswerList.Clear();
+        foreach (var question in Questions)
+        {
+            var answer = new Answer(
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:166]: Should be new Result

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs (hunk 3 above)
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:159]: Maybe make async

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs [hunk 5]
```diff
@@ -27,4 +98,125 @@ public bool IsLastPage
...
+            currentAnswerList.Add(answer);
+
+            // currently we send in the answers each time
+            _client.StoreResultFromQuestion(answer);
+        }
+    }
+
+    private void FillSavedAnswerToQuestions()
+    {
+        var answers = _results[_resultIdx];
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:182]: Naming

**Review by NicolajRJensen — APPROVED** [review]
GTM

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs [hunk 6]
```diff
@@ -1,21 +1,90 @@
...
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Windows.Input;
+using Model.Answer;
+using Model.FrontEndAPI;
+using Model.Question;
+using Model.Result;
+using Model.Survey;
 
 namespace scivu.ViewModels;
```
Comment (NicolajRJensen) [inline src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs:12]: Fix ordering of using/Namespace

File: src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs (hunk 3 above)
Comment (jonasreholt) [reply src/scivu/scivu/ViewModels/SurveyTakeViewModel.cs]: Won't do. Let's look at it if it becomes a problem

**Review by NicolajRJensen — APPROVED** [review]

**Pull request merged**